File: Java/Desafios-sala/src/Listas.java
import java.util.ArrayList;

class Funcionario{
    public double calcularBonus(){
        return 500.0;
    }
}

class Gerente extends Funcionario{
    public double calcularBonus(){
        return 2000.0;
    }
}

class Operario extends Funcionario{
    public double calcularBonus(){
        return 1000.0;
    }
}

public class Listas {
    public static void main(String[] args){
        ArrayList<Funcionario> lista = new ArrayList<>();
        lista.add(new Gerente());
        int n = lista.size();
        for (int i = 0; i < n; i++){
            System.out.println(lista.get(i).calcularBonus());
        }

        for (Funcionario i : lista){
            System.out.println(i.calcularBonus());
        }
    }    
}

File: Python/Desafios-sala/Listas.py
class Funcionario:
    def calcularBonus(self):
        return 500.0

class Gerente(Funcionario):
    def calcularBonus(self):
        return 2000.0
    
class Operario(Funcionario):
    def calcularBonus(self):
        return 1000.0
    
func1 = Funcionario()
func2 = Gerente()
func3 = Operario()
lista = [func1, func2, func3]

print(func1.calcularBonus())
print(func2.calcularBonus())
print(func3.calcularBonus())
print()

for f in lista:
    print(f"Bonus: R$ {f.calcularBonus()}") 